lib/gtfs/components/ShowAllRoutesOnMapFilter.js

```diff
@@ -1,30 +1,52 @@
 // @flow
 
+import Icon from '@conveyal/woonerf/components/icon'
 import React, {Component} from 'react'
 import {Checkbox} from 'react-bootstrap'
 
 import * as shapesActions from '../actions/shapes'
+import {getComponentMessages} from '../../common/util/config'
 
+import type {FetchStatus} from '../../types'
 import type {Props as ContainerProps} from '../containers/ShowAllRoutesOnMapFilter'
 
 type Props = ContainerProps & {
+  fetchStatus: FetchStatus,
   showAllRoutesOnMap: boolean,
   toggleShowAllRoutesOnMap: typeof shapesActions.toggleShowAllRoutesOnMap
 }
 
 export default class ShowAllRoutesOnMapFilter extends Component<Props> {
+  messages = getComponentMessages('ShowAllRoutesOnMapFilter')
+
+  onChange = () => {
+    const {toggleShowAllRoutesOnMap, version} = this.props
+    toggleShowAllRoutesOnMap(version.namespace)
+  }
+
   render () {
     const {
+      fetchStatus,
       showAllRoutesOnMap,
-      toggleShowAllRoutesOnMap
+      version
     } = this.props
-
+    // Disable if greater than one million records. We don't want to break
+    // Data Tools because of this shapes GraphQL fetch. This is a somewhat
```

Comment:
Why is this based on individual shapes instead on individual routes or route patterns that we plot somewhere else?

Reply:
Because this is just showing the shapes on a map, where the other is showing patterns that are selectable in the editor. The editor piece isn't really working though. We should see if we can consolidate these though whenever we fix the editor piece.